### PR TITLE
Move endpoint generation to function and add documentation explaining…

### LINKED ISSUE #1152: Add a comment saying that sends the requests to `jobs/yaptide` if `platform` = yaptide.
Add a comment saying that sends the requests to `jobs/yaptide` if `platform` = yaptide.
We will be vey surprised if for whatever aestetic reasons we change the platform name, not changing the endpoint and this will stop working.

The connection between platform name and endpoint name is not written anywahere

_Originally posted by @grzanka in https://github.com/yaptide/ui/pull/1148#discussion_r1290214223_

### PR DESCRIPTION
**Pull request by Derstilon**
… this function.
close #1152

### CODE REVIEW

**Comment by grzanka** [comment]
image

**Review by grzanka — CHANGES_REQUESTED** [review]
still doesn;t work for batch

**Comment by grzanka** [comment]
Still not working

These are GET requests for job status:
```
https://yap-dev.c3.plgrid.pl:8443/jobs/?job_id=20230908-144530-670ae0b8-c30e-40e3-af8d-1eee098a0b81DIRECT
```
Shouldn't rather be:
```
https://yap-dev.c3.plgrid.pl:8443/jobs?job_id=20230908-144530-670ae0b8-c30e-40e3-af8d-1eee098a0b81DIRECT
```

?

image

File: src/services/ShSimulatorService.tsx [hunk 1]
```diff
@@ -146,6 +146,25 @@ const ShSimulation = ({ children }: GenericContextProviderProps) => {
...
+	 * @see PlatformType
+	 * @param info - The simulation info object.
+	 * @returns The endpoint string.
+	 */
+	const getEndpointFromSimulationInfo = (info: SimulationInfo) => {
+		const platform =
+			`jobs/${info.metadata.platform.toLowerCase()}` as `jobs/${Lowercase<PlatformType>}`;
+
+		if (platform in ['jobs/direct', 'jobs/batch']) return platform;
+		console.error(`Simulation platform is unknown: ${info.metadata.platform}`);
```
Comment (grzanka) [inline src/services/ShSimulatorService.tsx:163]: It happens that this code fails (i.e. batch simulation + manual refresh of job status), the code fails with:

Simulation platform is unknown: BATCH

That suggests tha the logic above is somehow wrong,

Comment (Derstilon) [reply src/services/ShSimulatorService.tsx:163]: `in` is a keyword for objects mistakenly used here to search array
I will replace it with includes
image

**Review by grzanka — APPROVED** [review]

**Pull request merged**